monitors/rootmonitor/include/eudaq/ROOTMonitor.hh

```diff
@@ -0,0 +1,53 @@
+#include "eudaq/Monitor.hh"
+#include "eudaq/ROOTMonitorWindow.hh"
+
+#ifndef __CINT__
+# include "TApplication.h"
+# include "RQ_OBJECT.h"
+#endif
+
+class TH1D;
+class TGraph;
+
+namespace eudaq {
+  class ROOTMonitor : public Monitor {
+    static constexpr const char* NAME = "eudaq::ROOTMonitor";
+    RQ_OBJECT(NAME)
+  public:
+    ROOTMonitor(const std::string & name, const std::string & title, const std::string & runcontrol);
+
+    void LoadRAWFileAsync(const char* path);
+
+    void DoInitialise() override;
+    void DoConfigure() override;
+    void DoStartRun() override;
+    void DoStopRun() override;
+    void DoTerminate() override;
+    void DoReset() override;
+    void DoReceive(eudaq::EventSP ev) override;
+
+    virtual void AtInitialisation() {}
+    virtual void AtConfiguration() {}
+    virtual void AtRunStart() {}
+    virtual void AtRunStop() {}
+    virtual void AtEventReception(eudaq::EventSP ev) = 0;
```

Comment:
`AtEventReception(eudaq::EventSP ev) ` is required be to implemented in ROOT macro by end user. Am I wrong?

Reply:
You may find a (messy...) example (along with the `AtConfigure` part in https://github.com/forthommel/eudaq/blob/cmstotem/user/cmstotem/module/src/SampicMonitor.cc#L108-L186

Reply:
And indeed, I left this base class as a pure virtual object to ensure the monitor actually makes sense (and monitors something... :-) )